WeakAuras/SubRegionTypes/Tick.lua

```diff
@@ -0,0 +1,182 @@
+  },
+  tick_width = {
+    display = L["Width"],
+    setter = "SetTickWidth",
+    type = "number",
+    min = 0,
+    bigStep = 1,
+    default = 2,
+  },
+  tick_height = {
+    display = L["Height"],
+    setter = "SetTickHeight",
+    type = "number",
+    min = 0,
+    bigStep = 1,
+    default = 30,
+    --hidden = function() return data.automatic_height end,
+  },
+}
+
+local auraBarTrueLeft = {
+  ["HORIZONTAL"] = "LEFT",
+  ["HORIZONTAL_INVERSE"] = "RIGHT",
+  ["VERTICAL"] = "TOP",
+  ["VERTICAL_INVERSE"] = "BOTTOM",
+}
+
+local auraBarTrueLeftInverse = {
+  ["HORIZONTAL"] = "RIGHT",
+  ["HORIZONTAL_INVERSE"] = "LEFT",
+  ["VERTICAL"] = "BOTTOM",
+  ["VERTICAL_INVERSE"] = "TOP",
+}
+
+local function create()
+  local subRegion = CreateFrame("FRAME", nil, UIParent)
+  subRegion.texture = subRegion:CreateTexture()
+  subRegion.texture:SetDrawLayer("ARTWORK", 3)
+  subRegion.texture:SetAllPoints(subRegion)
+  return subRegion
+end
+
+local function onAcquire(subRegion)
+  subRegion:Show()
+end
+
+local function onRelease(subRegion)
+  subRegion:Hide()
+end
+
+local funcs = {
+  ["Update"] = function(self)
+    self:SetTickPlacement(self.tick_placement)
+  end,
+  SetVisible = function(self, visible)
+    if visible then
+      self:Show()
+    else
+      self:Hide()
+    end
+  end,
+  SetTickColor = function(self, r, g, b, a)
+    self.texture:SetColorTexture(r, g, b, a or 1)
+  end,
+  SetTickPlacement = function(self, placement)
+    local offset, offsetx, offsety = self.tick_placement, 0, 0
+    local width = self.parentTrueWidth
+    if self.tick_placement_mode == "STATIC" then
+      local percent = string.match(placement, "(%d+)%%")
+      if percent then
+        offset = (tonumber(percent) / 100) * width
+      else
+        local pixels = width / (self.parent.state.duration or self.parent.state.total or 1)
+        offset = tonumber(placement) * pixels
+      end
+    end
+
+    if self.parentInverse ~= ((self.parentOrientation == "HORIZONTAL_INVERSE") or (self.parentOrientation == "VERTICAL")) then
+      offset = -offset
+    end
+    if (self.parentOrientation == "VERTICAL") or (self.parentOrientation == "VERTICAL_INVERSE") then
+      offsety = offset
+    else
+      offsetx = offset
+    end
+    local side = self.parentInverse and auraBarTrueLeftInverse or auraBarTrueLeft
+
+    self:ClearAllPoints()
+    self:SetPoint("CENTER", self.parent.bar, side[self.parentOrientation], offsetx, offsety)
+  end,
+  SetTickWidth = function(self, width)
+    if (self.parentOrientation == "VERTICAL") or (self.parentOrientation == "VERTICAL_INVERSE") then
+      self:SetHeight(width)
+    else
+      self:SetWidth(width)
+    end
+  end,
+  SetTickHeight = function(self, height)
+    if self.automatic_height then height = self.parentTrueHeight end
+    if (self.parentOrientation == "VERTICAL") or (self.parentOrientation == "VERTICAL_INVERSE") then
+      self:SetWidth(height)
+    else
+      self:SetHeight(height)
+    end
+  end,
```

Comment:
I made the conscious decision to keep width always the direction the bar is moving and height the perpendicular. In the aura bar's option changing orientation causes the height and width values to flip, but I don't see a clean way of replicating that behavior in a sub region.

Reply:
Then "width" and "height" are not the right words. Both in the user interface and in the code

There are two ways to do this:
Use a generic word, such as "Perpendicular", "Major/Minor", "Across" as the Width/Height

Adjust the naming to whatever the current orientation is. So if its horizontal, the names are height/width, whereas otherwise they are width/height.

For the code, I think widht/height aren't ideal either.